OpenRA.Mods.Common/Warheads/DamageWarhead.cs

```diff
@@ -34,13 +35,13 @@ public static object LoadVersus(MiniYaml yaml)
 				: new Dictionary<string, int>();
 		}
 
-		public int DamageVersus(ActorInfo victim)
+		public int DamageVersus(Actor victim)
 		{
-			var armor = victim.Traits.GetOrDefault<ArmorInfo>();
-			if (armor != null && armor.Type != null)
+			var armor = victim.TraitsImplementing<Armor>().Where(a => !a.IsTraitDisabled && a.Info.Type != null);
+			foreach (var a in armor)
 			{
 				int versus;
```

Comment:
Move the creation of this out from the foreach, with a default value of 100 and multiply it here to let more armors stack with each other, also move the return out from the foreach too.